include/mega/types.h

```diff
@@ -311,7 +311,42 @@ typedef list<HttpReqCommandPutFA*> putfa_list;
 // map a FileFingerprint to the transfer for that FileFingerprint
 typedef map<FileFingerprint*, Transfer*, FileFingerprintCmp> transfer_map;
 
-typedef deque<Transfer*> transfer_list;
+template <class T, class E>
+class deque_with_lazy_bulk_erase
```

Comment:
`types.h` is included in about other 25 header files. I think we should try to help the compiler a bit and move the definitions to the `utils.cpp` as other objects declared here (when they don't have a better target source file).